lib/swiss_village_directory/village.rb

```diff
@@ -1,15 +1,17 @@
+require 'bigdecimal/util'
+
 module SwissVillageDirectory
   class Village
     attr_reader :name, :zip_code, :one_digit_spare, :commune, :canton, :longitude, :latitude
 
     def initialize(name, zip_code, one_digit_spare, commune, canton, longitude, latitude)
       @name = name
-      @zip_code = zip_code.to_i
+      @zip_code = zip_code.to_s
       @one_digit_spare = one_digit_spare.to_i
       @commune = commune
       @canton = canton
-      @longitude = longitude.to_f
-      @latitude = latitude.to_f
+      @longitude = longitude.to_d
+      @latitude = latitude.to_d
```

Comment:
Are you sure about the decimals? I'd stick with floats because JSON cannot represent decimals. Decimal precision is over-engineering in my eyes.

Reply:
In this case it doesn't matter too much. I can switch back to floats